client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -70,20 +70,42 @@
     private final SynchronizerConfig synchronizerConfig;
     private final Serializer<ReaderGroupStateInit> initSerializer;
     private final Serializer<ReaderGroupStateUpdate> updateSerializer;
-    private final ClientFactory clientFactory;
     private final Controller controller;
-    private final ConnectionFactory connectionFactory;
+    private final SegmentMetadataClientFactory metaFactory;
     private final NotificationSystem notificationSystem = new NotificationSystem();
-    private final NotifierFactory notifierFactory = new NotifierFactory(notificationSystem, this::createSynchronizer);
+    private final StateSynchronizer<ReaderGroupState> synchronizer;
+    private final NotifierFactory notifierFactory;
+
+    
+    public ReaderGroupImpl(String scope, String groupName, SynchronizerConfig synchronizerConfig,
+                           Serializer<ReaderGroupStateInit> initSerializer, Serializer<ReaderGroupStateUpdate> updateSerializer,
+                           ClientFactory clientFactory, Controller controller, ConnectionFactory connectionFactory) {
+        Preconditions.checkNotNull(scope);
```

Comment:
Tip: `Preconditions.checkNotNull` returns the value of the agument if the check passes. So you can compact your code like:
`this.scope = Preconditions.checkNotNull(scope);`
(and so on)